R/auto.R

```diff
@@ -57,68 +77,108 @@ guess_next_impl <- function(version) {
-  remote_name <- get_remote_name(main_branch)
-
-  cli_h1("1. Creating a release branch and getting ready")
+  cli_h1("1. Wrapping up development")
 
-  # bump version on main branch to version set by user
-  # Avoid `bump_version()` to avoid showing `NEWS.md` at this stage,
-  # because it changes as we jump between branches.
-  update_news(which = which)
-  commit_version()
-
-  # switch to release branch and update cran-comments
-  release_branch <- create_release_branch(force)
-  switch_branch(release_branch)
-  update_cran_comments()
+  # Bump dev if needed, bail out if bumped
+  new_fledgeling <- bump_version_impl(fledgeling, "dev", no_change_behavior = "noop")
+  if (!identical(new_fledgeling, fledgeling)) {
+    return()
+  }
+  rm(new_fledgeling)
+
+  # Bump dev unconditionally, ignore resulting NEWS entry
+  bump_version_impl(
+    fledgeling,
+    which = "dev",
+    no_change_behavior = "bump",
+    edit = FALSE,
+    no_change_message = paste0(
+      "Initiated release of version ",
+      new_version,
+      " from version ",
+      fledgeling$version
+    )
+  )
 
-  # push main branch, bump to devel version and push again
-  push_to_new(remote_name, force)
-  switch_branch(main_branch)
+  cli_h1("2. Creating a release branch and getting ready")
 
-  cli_h1("2. Bumping main branch to dev version and updating NEWS")
-  # manual implementation of bump_version(), it doesn't expose `force` yet
-  bump_version_to_dev_with_force(force, which = "dev")
+  # regroup dev news
+  fledgeling <- merge_dev_news(fledgeling, new_version)
 
-  cli_h1("3. Opening Pull Request for release branch")
-  # switch to release branch and init pre_release actions
+  # switch to release branch and update cran-comments
+  release_branch <- create_release_branch(fledgeling, force)
   switch_branch(release_branch)
 
-  cli_alert("Opening draft pull request with contents from {.file cran-comments.md}.")
-  create_pull_request(release_branch, main_branch, remote_name, force)
+  write_fledgling(fledgeling)
+  update_cran_comments()
 
   edit_news()
   edit_cran_comments()
 
-  # user action items
-  cli_h1("4. User Action Items")
+  cli_h1("3. User Action Items")
   cli_div(theme = list(ul = list(color = "magenta")))
   cli_ul("Run {.code devtools::check_win_devel()}.")
   cli_ul("Run {.code rhub::check_for_cran()}.")
   cli_ul("Run {.code urlchecker::url_update()}.")
   cli_ul("Check all items in {.file cran-comments.md}.")
-  cli_ul("Convert {.file NEWS.md} from changelog format to release notes.")
-  cli_ul("Run {.code fledge::release()}.")
-  cli_end()
+  cli_ul("Review {.file NEWS.md}.")
+  cli_ul("Call {.code pre_release()}.")
 
   send_to_console("urlchecker <- urlchecker::url_update(); fledge:::bg_r(winbuilder = devtools::check_win_devel(quiet = TRUE), rhub = rhub::check_for_cran())")
+}
+
+pre_release_impl <- function(force) {
+  cli_h1("1. Opening Pull Request for release branch")
+
+  if (nrow(gert::git_status()) > 0) {
```

Comment:
add an explaining variable here?
```suggestion
  changes_made <- nrow(gert::git_status()) > 0
  if (changes_made) {
```